File: Models/DTO/OrderDTO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Models.DTO
{
    public class OrderDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool IsExecuted { get; set; }
        public DateTime Date { get; set; }
        public ProductInCartDTO Quantitys { get; set; }
        public int ProductID { get; set; }
    }
}

File: ClientApp/src/app/models/orderDTO.ts
import { DiscountDTO } from "./discountDTO";
import { ProductInCart } from "./productInCart";

export class OrderDTO{
    id: number;
    name: string;
    isExecuted: boolean;
    date:Date;
    products:Array<ProductInCart>
    discounts:Array<DiscountDTO>
}